### PR TITLE
Clean up src/client.jsx

### PR DESCRIPTION
**Pull request by martinpitt**

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -37,59 +40,42 @@ function podmanMonitor(name, method, args, callback, system) {
 }
 
 export function streamEvents(system, callback) {
-    return new Promise((resolve, reject) => {
-        podmanMonitor("libpod/events", "GET", {}, callback, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanMonitor("libpod/events", "GET", {}, callback, system)
+            .then(reply => JSON.parse(reply));
```
Comment (cockpituous) [inline src/client.js:44]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 2]
```diff
@@ -37,59 +40,42 @@ function podmanMonitor(name, method, args, callback, system) {
...
 export const renameContainer = (system, id, config) => podmanCall("libpod/containers/" + id + "/rename", "POST", config, system);
 
-export function createContainer(system, config) {
-    return new Promise((resolve, reject) => {
-        podmanCall("libpod/containers/create", "POST", {}, system, JSON.stringify(config))
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const createContainer = (system, config) => podmanJson("libpod/containers/create", "POST", {}, system, JSON.stringify(config));
```
Comment (cockpituous) [inline src/client.js:78]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 3]
```diff
@@ -112,11 +98,7 @@ export function execContainer(system, id) {
         Cmd: ["/bin/sh"],
     };
 
-    return new Promise((resolve, reject) => {
-        podmanCall("libpod/containers/" + id + "/exec", "POST", {}, system, JSON.stringify(args))
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanJson("libpod/containers/" + id + "/exec", "POST", {}, system, JSON.stringify(args));
```
Comment (cockpituous) [inline src/client.js:101]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 4]
```diff
@@ -147,66 +129,44 @@ function parseImageInfo(info) {
...
-                                }
-                                resolve(images);
-                            })
-                            .catch(reject);
-                })
-                .catch(reject);
-    });
+    const options = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
```
Comment (cockpituous) [inline src/client.js:134]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 5]
```diff
@@ -147,66 +129,44 @@ function parseImageInfo(info) {
...
+    return podmanJson("libpod/images/json", "GET", options, system)
+            .then(reply => {
+                const images = {};
+                const promises = [];
+
+                for (const image of reply || []) {
```
Comment (cockpituous) [inline src/client.js:140]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 6]
```diff
@@ -147,66 +129,44 @@ function parseImageInfo(info) {
...
-    return new Promise((resolve, reject) => {
-        const options = {
-            force,
-        };
-        podmanCall("libpod/images/" + id, "DELETE", options, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const delImage = (system, id, force) => podmanJson("libpod/images/" + id, "DELETE", { force }, system);
```
Comment (cockpituous) [inline src/client.js:163]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 7]
```diff
@@ -147,66 +129,44 @@ function parseImageInfo(info) {
...
 
 export const untagImage = (system, id, repo, tag) => podmanCall("libpod/images/" + id + "/untag", "POST", { repo, tag }, system);
 
 export function pullImage(system, reference) {
     return new Promise((resolve, reject) => {
-        const options = {
-            reference,
-        };
-        podmanCall("libpod/images/pull", "POST", options, system)
+        podmanCall("libpod/images/pull", "POST", { reference }, system)
```
Comment (cockpituous) [inline src/client.js:169]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 8]
```diff
@@ -223,20 +183,8 @@ export function pullImage(system, reference) {
...
 }
 
-export function pruneUnusedImages(system) {
-    return new Promise((resolve, reject) => {
-        podmanCall("libpod/images/prune?all=true", "POST", {}, system).then(resolve)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const pruneUnusedImages = system => podmanJson("libpod/images/prune?all=true", "POST", {}, system);
```
Comment (cockpituous) [inline src/client.js:186]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 9]
```diff
@@ -223,20 +183,8 @@ export function pullImage(system, reference) {
...
 
-export function imageHistory(system, id) {
-    return new Promise((resolve, reject) => {
-        podmanCall(`libpod/images/${id}/history`, "GET", {}, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const imageHistory = (system, id) => podmanJson(`libpod/images/${id}/history`, "GET", {}, system);
```
Comment (cockpituous) [inline src/client.js:188]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 10]
```diff
@@ -37,11 +37,8 @@ function podmanMonitor(name, method, args, callback, system) {
 }
 
 export function streamEvents(system, callback) {
-    return new Promise((resolve, reject) => {
-        podmanMonitor("libpod/events", "GET", {}, callback, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    return podmanMonitor("libpod/events", "GET", {}, callback, system)
+            .then(reply => JSON.parse(reply));
```
Comment (cockpituous) [inline src/client.js:41]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-034759-cb60c9a8-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 11]
```diff
@@ -147,66 +132,50 @@ function parseImageInfo(info) {
...
+    const options = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanCall("libpod/images/json", "GET", options, system)
+            .then(reply => {
+                const immages = JSON.parse(reply);
+                const images = {};
+                const promises = [];
+
+                for (const image of immages || []) {
```
Comment (cockpituous) [inline src/client.js:144]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-034759-cb60c9a8-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js (hunk 1 above)
Comment (cockpituous) [inline src/client.js:44]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-052110-5d572290-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js (hunk 5 above)
Comment (cockpituous) [inline src/client.js:140]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-052110-5d572290-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: src/client.js [hunk 12]
```diff
@@ -147,66 +126,44 @@ function parseImageInfo(info) {
...
-    });
+    const options = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanJson("libpod/images/json", "GET", options, system)
+            .then(reply => {
+                const images = {};
+                const promises = [];
+
+                for (const image of reply || []) {
```
Comment (cockpituous) [inline src/client.js:137]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-053157-361a084c-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (martinpitt) [reply src/client.js:137]: I keep wondering what this complains about -- but possibly the `|| []`? That is indeed unnecessary, let me try..

**Review by jelly — APPROVED** [review]
One day we will understand promises :)

Nice cleanups!

**Comment by jelly** [comment]
Rawhide broke due to Python 3.12
```
    Package "python3-3.11.4-1.fc39.x86_64" is already installed.
    Package "cockpit-podman-71-1.20230706054442288477.pr1339.32.gff52e07.fc39.noarch" is already installed.
    Problem 1: problem with installed package 
      - package python3-dnf-plugin-tracer-4.1.0-1.fc39.noarch requires python(abi) = 3.11, but none of the providers can be installed
      - package python3-3.11.4-1.fc39.i686 requires libpython3.11.so.1.0, but none of the providers can be installed
      - package python3-3.11.4-1.fc39.i686 requires python3-libs(x86-32) = 3.11.4-1.fc39, but none of the providers can be installed
      - python3-libs-3.11.4-1.fc39.i686 has inferior architecture
      - cannot install both python3-libs-3.11.4-1.fc39.x86_64 and python3-libs-3.12.0~b3-2.fc39.x86_64
      - package python3-3.12.0~b3-2.fc39.x86_64 requires python3-libs(x86-64) = 3.12.0~b3-2.fc39, but none of the providers can be installed
      - cannot install both python3-3.11.4-1.fc39.x86_64 and python3-3.12.0~b3-2.fc39.x86_64
      - cannot install the best candidate for the job
     Problem 2: problem with installed package 
      - package python3-dnf-plugins-extras-common-4.1.0-1.fc39.noarch requires python(abi) = 3.11, but none of the providers can be installed
      - package python3-3.11.4-1.fc39.i686 requires libpython3.11.so.1.0, but none of the providers can be installed
      - package python3-3.11.4-1.fc39.i686 requires python3-libs(x86-32) = 3.11.4-1.fc39, but none of the providers can be installed
      - package python3-libs-3.11.4-1.fc39.i686 obsoletes python3.11-libs provided by python3.11-libs-3.11.4-2.fc39.x86_64
      - problem with installed package 
      - package sudo-python-plugin-1.9.13-4.p2.fc39.x86_64 requires libpython3.11.so.1.0()(64bit), but none of the providers can be installed
      - cannot install both python3-libs-3.11.4-1.fc39.x86_64 and python3-libs-3.12.0~b3-2.fc39.x86_64
      - package python3-3.12.0~b3-2.fc39.x86_64 requires python3-libs(x86-64) = 3.12.0~b3-2.fc39, but none of the providers can be installed
      - package python3-libvirt-9.5.0-1.fc39.x86_64 requires python(abi) = 3.12, but none of the providers can be installed
      - cannot install the best candidate for the job
      - cannot install both python3-3.11.4-1.fc39.x86_64 and python3-3.12.0~b3-2.fc39.x86_64
```

**Pull request merged**